### PR TITLE
chore: ☂️ Migrate GenerateCodegenSchemaTaskTest.kt to AssertJ

### PR DESCRIPTION
**Pull request by omer-ayhan**
## Summary:
Issue: https://github.com/facebook/react-native/issues/45596

## Changelog:
[INTERNAL] [CHANGED] - Migrate `GenerateCodegenSchemaTaskTest.kt` to AssertJ testing library

## Test Plan:
Run `./gradlew -p packages/gradle-plugin test`

### CODE REVIEW

**Comment by omer-ayhan** [comment]
hello @cortinico 👋 this PR is ready for a review🚀

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt [hunk 1]
```diff
@@ -65,14 +66,10 @@ class GenerateCodegenSchemaTaskTest {
...
-    assertEquals(
-        setOf(
-            "**/build/**/*",
-        ),
-        task.jsInputFiles.excludes)
-    assertEquals(1, task.jsInputFiles.files.size)
-    assertEquals(setOf(File(jsRootDir, "afolder/includedfile.js")), task.jsInputFiles.files)
+    assertThat(task.jsInputFiles.dir).isEqualTo(jsRootDir)
+    assertThat(task.jsInputFiles.excludes).isEqualTo(setOf("**/build/**/*"))
+    assertThat(task.jsInputFiles.files).hasSize(1)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt:71]: This is unnecessary now 👍

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt [hunk 2]
```diff
@@ -104,8 +101,8 @@ class GenerateCodegenSchemaTaskTest {
 
     task.wipeOutputDir()
 
-    assertTrue(File(tempFolder.root, "output").exists())
-    assertEquals(0, File(tempFolder.root, "output").listFiles()?.size)
+    assertThat(File(tempFolder.root, "output").exists()).isTrue()
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt:104]: `.exists()).isTrue()` -> `.exists()`

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt [hunk 3]
```diff
@@ -138,7 +135,7 @@ class GenerateCodegenSchemaTaskTest {
 
     task.setupCommandLine()
 
-    assertEquals(
+    assertThat(task.commandLine.toMutableList()).isEqualTo(
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt:138]:
```suggestion
    assertThat(task.commandLine).containsExactly(
        "node",
        "--verbose",
        File(codegenDir, "lib/cli/combine/combine-js-to-schema-cli.js").toString(),
        "--platform",
        "android",
        "--exclude",
        "NativeSampleTurboModule",
        File(outputDir, "schema.json").toString(),
        jsRootDir.toString(),
    )
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt [hunk 4]
```diff
@@ -171,7 +168,7 @@ class GenerateCodegenSchemaTaskTest {
...
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt:171]: same as above

**Review by cortinico — CHANGES_REQUESTED** [review]
Please include the suggested changes

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt (hunk 1 above)
Comment (omer-ayhan) [reply packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt:71]: should I delete those assertions? Which means you don't need assertion for `task.jsInputFiles`?

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt [hunk 5]
```diff
@@ -33,11 +33,12 @@ class GenerateCodegenSchemaTaskTest {
...
-    assertEquals(setOf("**/*.js", "**/*.ts"), task.jsInputFiles.includes)
-    assertEquals(2, task.jsInputFiles.files.size)
-    assertEquals(
-        setOf(File(jsRootDir, "file.js"), File(jsRootDir, "file.ts")), task.jsInputFiles.files)
+    assertThat(task.jsInputFiles.dir).isEqualTo(jsRootDir)
+    assertThat(task.jsInputFiles.includes).isEqualTo(setOf("**/*.js", "**/*.ts"))
+    assertThat(task.jsInputFiles.files).hasSize(2)
+    assertThat(task.jsInputFiles.files).containsExactlyInAnyOrder(
+        File(jsRootDir, "file.js"), File(jsRootDir, "file.ts")
+    )
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt:41]:
```suggestion
    assertThat(task.jsInputFiles.dir).isEqualTo(jsRootDir)
    assertThat(task.jsInputFiles.includes).isEqualTo(setOf("**/*.js", "**/*.ts"))
    assertThat(task.jsInputFiles.files).containsExactlyInAnyOrder(
        File(jsRootDir, "file.js"), File(jsRootDir, "file.ts")
    )
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt (hunk 1 above)
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt:71]:
```suggestion
    assertThat(task.jsInputFiles.dir).isEqualTo(jsRootDir)
    assertThat(task.jsInputFiles.excludes).isEqualTo(setOf("**/build/**/*"))
```

Comment (cortinico) [reply packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GenerateCodegenSchemaTaskTest.kt:71]: I meant only the size assertion

**Review by cortinico — APPROVED** [review]

**Pull request closed without merging**